From bda688c4b0cda4bd20ccf1d771f1f4285a9adf27 Mon Sep 17 00:00:00 2001
From: Zong Li <zong.li@sifive.com>
Date: Wed, 30 Jun 2021 23:23:47 +0800
Subject: [PATCH 05/16] riscv: sifive: fu740: Support i2c in spl

Enable SPL_I2C_SUPPORT for fu740, and add 'u-boot,dm-spl' property in
i2c node.

Signed-off-by: Zong Li <zong.li@sifive.com>
Reviewed-by: Leo Yu-Chi Liang <ycliang@andestech.com>
---
 arch/riscv/cpu/fu740/Kconfig          | 1 +
 arch/riscv/dts/fu740-c000-u-boot.dtsi | 4 ++++
 2 files changed, 5 insertions(+)

diff --git a/arch/riscv/cpu/fu740/Kconfig b/arch/riscv/cpu/fu740/Kconfig
index 1dc052b..8e54310 100644
--- a/arch/riscv/cpu/fu740/Kconfig
+++ b/arch/riscv/cpu/fu740/Kconfig
@@ -37,3 +37,4 @@ config SIFIVE_FU740
 	imply PWM_SIFIVE
 	imply DM_I2C
 	imply SYS_I2C_OCORES
+	imply SPL_I2C_SUPPORT
diff --git a/arch/riscv/dts/fu740-c000-u-boot.dtsi b/arch/riscv/dts/fu740-c000-u-boot.dtsi
index a5d0688..a6f7a08 100644
--- a/arch/riscv/dts/fu740-c000-u-boot.dtsi
+++ b/arch/riscv/dts/fu740-c000-u-boot.dtsi
@@ -95,6 +95,10 @@
 	u-boot,dm-spl;
 };
 
+&i2c0 {
+	u-boot,dm-spl;
+};
+
 &eth0 {
 	assigned-clocks = <&prci PRCI_CLK_GEMGXLPLL>;
 	assigned-clock-rates = <125125000>;
-- 
2.7.4

